pytorch_tabnet/tab_model.py

```diff
@@ -217,6 +270,59 @@ def fit(self, X_train, y_train, X_valid=None, y_valid=None, loss_fn=None,
         # compute feature importance once the best model is defined
         self._compute_feature_importances(train_dataloader)
 
+    def save_model(self, path):
+        """
+        Saving model with two distinct files.
+        """
+        saved_params = {}
+        for key, val in self.get_params().items():
+            if isinstance(val, type):
+                # Don't save torch specific params
+                continue
+            else:
+                saved_params[key] = val
+
+        # Create folder
+        Path(path).mkdir(parents=True, exist_ok=True)
+
+        # Save models params
+        with open(Path(path).joinpath("model_params.json"), "w", encoding="utf8") as f:
+            json.dump(saved_params, f)
+
+        # Save state_dict
+        torch.save(self.network.state_dict(), Path(path).joinpath("network.pt"))
```

Comment:
Not sure about having non-unique filename here :(

Also, when using Path, you can just do ```

 Path(path) / "network.pt

```

Reply:
Ok, after reading it again, looks like unique name are fine.
However, I feel like this would be much better to 
- create model_params.json
- create network.pt
- zip them together, and use the path provided as zip name

Then to reload, read the files inside the zip and init them.

That way, the model needs only one file to be saved.

Also I think there is an Sklearn method to get all __init__ params.

Reply:
well the name of the folder is given by the user so it's like any other file saving.

Reply:
yes I was thinking about zipping everything to a single file too, will have a look on this when I have time.

Reply:
done!